lib/qesapdeployment.pm

```diff
@@ -826,39 +827,91 @@ sub qesap_az_vnet_peering {
 
     Performs peering between the cluster and IBS mirror on Azure.
 
-=over 3
+=over 2
 
 =item B<SOURCE_GROUP> - resource group of source
 
-=item B<SOURCE_VNET> - vnet of source
-
 =item B<TARGET_GROUP> - resource group of target
 
-=item B<TARGET_VNET> - vnet of target
-
 =back
 =cut
 
 sub qesap_delete_az_peering {
     my (%args) = @_;
-    croak 'Missing mandatory source_group argument' unless $args{source_group};
     croak 'Missing mandatory target_group argument' unless $args{target_group};
 
-    $args{source_vnet} = qesap_get_vnet($args{source_group});
     $args{target_vnet} = qesap_get_vnet($args{target_group});
 
-    my $peering_name = "$args{source_vnet}-$args{target_vnet}";
-    my $source_cmd = "az network vnet peering delete --resource-group $args{source_group} --vnet-name $args{source_vnet} -n $peering_name";
-    my $target_cmd = "az network vnet peering delete --resource-group $args{target_group} --vnet-name $args{target_vnet} -n $peering_name";
+    my $peering_name = qesap_get_peering_name();
+    if (!$peering_name) {
+        record_info("NO PEERING", "No peering to be destroyed!");
+        return;
+    }
 
     record_info("Attempting peering destruction");
-    my $source_ret = script_run($source_cmd);
+    my $source_ret = 0;
+    if ($args{source_group}) {
+        $args{source_vnet} = qesap_get_vnet($args{source_group});
+        my $source_cmd = "az network vnet peering delete --resource-group $args{source_group} --vnet-name $args{source_vnet} -n $peering_name";
+        $source_ret = script_run($source_cmd);
+    }
+    my $target_cmd = "az network vnet peering delete --resource-group $args{target_group} --vnet-name $args{target_vnet} -n $peering_name";
     my $target_ret = script_run($target_cmd);
+
     if ($source_ret == 0 && $target_ret == 0) {
         record_info("Peering deletion SUCCESS", "The peering was successfully destroyed");
         return;
     }
-    record_info("Peering destruction FAIL", "There may be leftover peering connections, please check");
+    record_soft_failure("Peering destruction FAIL: There may be leftover peering connections, please check - jira#7487");
 }
 
+=head3 qesap_get_peering_name
+
+    Returns the peering name for the current job, or
+    empty string if a peering doesn't exist
+
+=cut
+
+sub qesap_get_peering_name {
+    my $job_id = get_current_job_id();
+    my $json_response = script_output("az network vnet peering list -g dsdm0we --vnet-name vnet-dsdm0we");
```

Comment:
For a next PR: it is an `az` specific implementation. We should add `az` in the function name

Reply:
For a next PR: I think that we can create a constant variable in this file to store `dsdm0we`

Reply:
we may possibly also want to export this, if it is to be used in the future